addons/misc/functions/fnc_onDeathZeus.sqf

```diff
@@ -0,0 +1,184 @@
+#include "script_component.hpp"
+/*/////////////////////////////////////////////////
+Author: Landric
+			   
+File: fnc_onDeathZeus.sqf
+Parameters: pos, unit
+Return: none
+
+Activates one or more configurable effects when the passed unit (or object) dies
+
+*///////////////////////////////////////////////
+
+
+
+params [["_pos",[0,0,0],[[]],3], ["_unit",objNull,[objNull]]];
+
+if(isNull _unit) exitWith { hint "Must be placed on a unit or object"; };
+
+private _onConfirm =
+{
+	params ["_dialogResult","_in"];
+
+	private _expectedParams = [
+		"_hint_killer",
+		"_hint_all",
+		"_hint_zeus",
+		"_deleteRemains",
+		"_explosion",
+		"_customCode",
+		"_codeTarget"
+	];
+
+	if (isClass(configFile >> "CfgPatches" >> "crowsEW_main")) then {
+		_expectedParams insert [5, ["_emp", "_sound_killer", "_sound_all"]];
+	};
+
+	_dialogResult params _expectedParams;
+	_in params ["_pos", ["_unit",objNull,[objNull]]];
+
+	if(isNull _unit) exitWith { hint "Something has gone wrong!"; diag_log "CrowsZA-onDeathZeus: unit became invalid after selection";};
+
+	if(
+		_hint_killer isEqualTo "" &&
+		{_hint_all isEqualTo "" &&
+		{_hint_zeus isEqualTo "" &&
+		{!_deleteRemains &&
+		{_explosion == 0 &&
+		{_customCode isEqualTo "" &&
+		{isNil "_emp" || {!_emp}} &&
+		{isNil "_sound_killer" || {_sound_killer isEqualTo ""}} &&
+		{isNil "_sound_all" || {_sound_all isEqualTo ""}}}}}}}
+	) exitWith { hint "Nothing set - no action taken!" };
+
+
+	private _unitList = missionNamespace getVariable ["crowza_onKilledEHUnits", []];
```

Comment:
should probably use macro ``GETMVAR()`` and ``SETMVAR()`` like used in EWAR sound modules. If they don't exist just add them to the main script_macros file like here: https://github.com/Crowdedlight/Crows-Electronic-Warfare/blob/develop/addons/main/script_macros.hpp#L11-L12

Reply:
Also please use `QGVAR()` macro for the naming of the variable. 
```
getVariable [QGVAR(onKilledEHUnits), ...
```

Reply:
Also consider if naming it ``OnKilledModuleEHUnits`` would make more sense. To avoid any confusion with the games native ``onKilled`` events.